docs/operate/control/headless-app.md

```diff
@@ -5,5 +5,532 @@ weight: 30
+
+    std::string base_name("viam_base");
+
+    cout << "Getting base: " << base_name << endl;
+    std::shared_ptr<Base> base;
+    try {
+        base = machine->resource_by_name<Base>(base_name);
+
+        move_in_square(base);
+
+    } catch (const std::exception& e) {
+        cerr << "Failed to find " << base_name << ". Exiting." << endl;
+        throw;
+    }
+    return EXIT_SUCCESS;
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{< /expand >}}
+
+{{< expand "Example Python code for turning on a fan based on sensor readings" >}}
+
+The following example from [Automate air filtration with air quality sensors](https://codelabs.viam.com/guide/air-quality/index.html?index=..%2F..index#0) uses both the sensor API's [GetReadings](/dev/reference/apis/components/sensor/#getreadings) command as well as custom functionality with the generic [DoCommand](/appendix/apis/components/generic/#docommand).
+
+```python {class="line-numbers linkable-line-numbers"}
+import asyncio
+import os
+from dotenv import load_dotenv
+from viam.logging import getLogger
+from viam.robot.client import RobotClient
+from viam.components.sensor import Sensor
+from viam.components.generic import Generic
+
+load_dotenv()
+LOGGER = getLogger(__name__)
+
+robot_api_key = os.getenv('ROBOT_API_KEY') or ''
+robot_api_key_id = os.getenv('ROBOT_API_KEY_ID') or ''
+robot_address = os.getenv('ROBOT_ADDRESS') or ''
+
+# Define the sensor and plug names from the Viam app CONFIGURE tab
+sensor_name = os.getenv("SENSOR_NAME", "")
+plug_name = os.getenv("PLUG_NAME", "")
+
+async def connect():
+    opts = RobotClient.Options.with_api_key(
+        api_key=robot_api_key,
+        api_key_id=robot_api_key_id
+    )
+    return await RobotClient.at_address(robot_address, opts)
+
+async def main():
+    machine = await connect()
+
+    pms_7003 = Sensor.from_robot(machine, sensor_name)
+    kasa_plug = Generic.from_robot(machine, plug_name)
+
+    # Define unhealthy thresholds
+    unhealthy_thresholds = {
+        'pm2_5_atm': 35.4,
+        'pm10_atm': 150
+    }
+
+    while True:
+        readings = await pms_7003.get_readings()
+        # Check if any of the PM values exceed the unhealthy thresholds
+        if any(readings.get(pm_type, 0) > threshold for pm_type, threshold in unhealthy_thresholds.items()):
+            LOGGER.info('UNHEALTHY.')
+            await kasa_plug.do_command({"toggle_on": []})
+        else:
+            LOGGER.info('HEALTHY!')
+            await kasa_plug.do_command({"toggle_off": []})
+
+        # wait before checking again
+        await asyncio.sleep(10)
+
+if __name__ == '__main__':
+    asyncio.run(main())
+```
+
+{{< /expand >}}
+
+## Run your script
+
+You can run your script manually from your terminal, or you can automatically run it each time your machine starts.
+
+### Run your script manually
+
+To run your script on a laptop or desktop, or to run it on your machine's SBC manually for testing purposes, execute the following command in a terminal:
```

Comment:
I think since they can run it anywhere (even in the cloud) we should simplify.
```suggestion
To run your script, execute the following command in a terminal:
```

Reply:
I meant for this to create distinction between this section and the run as process section. They can't really run as a process on a laptop or desktop (or at least most people wouldn't for practical reasons).